### DISCUSSION TITLE
I'm confused why NFS provisioner outperforms Jiva

### DISCUSSION THREAD

**Discussion by madalinignisca** [openebs/openebs#3626] [Q&A] [answered]
Hello,

I have this scenario, deployed on dedicated servers, nvme disks in raid 1.

Installed only Hostpath, Jiva and Nfs.
Jiva uses 3 Hostpath replicas, Nfs uses Jiva.

Using a Jiva PV, IOPS are very slow.
Using Nfs, IOPS are very high, ~40% of direct NVMe.

Here are screenshots:

<img width="774" alt="Screenshot 2023-03-23 at 07 14 39" src="https://user-images.githubusercontent.com/619561/227143265-d71bbe10-e8aa-4437-af60-a9c7f6705e09.png">

![twitter_Fr4l1-cWIAIoqFn](https://user-images.githubusercontent.com/619561/227143446-0cff1a9c-a4de-45a1-a3ab-372b0ecd622a.jpg)

Any idea why is direct Jiva usage so slow?

I did benchmarks with Longhorn, which gives identical results on normal or nfs mode, performance being somewhere at the middle between Jiva and Nfs over Jiva.

**Comment by avishnu** [comment, marked as answer]
Thanks for being a user of OpenEBS !!
I'd like to suggest Mayastor as the alternative to Jiva.
Mayastor is a replicated storage solution (just like Jiva) but it's very fast as compared to Jiva.
Mayastor is under active development and support, unlike Jiva.

https://github.com/openebs/mayastor
https://mayastor.gitbook.io/introduction/

**Reply by madalinignisca** [reply]
I did tested occasionally Mayastor, but the CPU usage is not acceptable. The cost for CPU is highest in cloud and dedicated computing.
